src/openrct2/ride/TrackData.h

```diff
@@ -67,10 +54,29 @@ constexpr const dodgems_track_size DodgemsTrackSize(track_type_t type)
         return { 4, 4, 59, 123 };
     return { 0, 0, 0, 0 };
 }
-extern const track_type_t TrackElementMirrorMap[TrackElemType::Count];
-
-extern const uint32_t TrackHeightMarkerPositions[TrackElemType::Count];
 
-extern const uint8_t TrackSequenceElementAllowedWallEdges[TrackElemType::Count][16];
+struct TrackElementDescriptor
+{
+    rct_track_coordinates Coordinates;
+
+    rct_preview_track* Block;
```

Comment:
For OpenLoco i modified this structure to return a `std::vector<rct_preview_track>` it makes things a lot simpler to handle but will require a lot of careful adjustment to the code to implement this.